### ISSUE TITLE
`no-missing-end-of-source-newline` gives unexpected errors and strange auto fixes

### ISSUE THREAD

**Issue by Xotic750** [stylelint/stylelint#3428]
> Clearly describe the bug

When running against our Vue js project, we get several of these in `.vue` files.

```
Modules/Calendar/Components/EventMenuHeader.vue
 18:38  ✖  Unexpected missing end-of-source newline   no-missing-end-of-source-newline
 23:37  ✖  Unexpected missing end-of-source newline   no-missing-end-of-source-newline
```

18: `<span style="margin-right: -3px;">(</span>`
23: `<span style="margin-left: -3px;">)</span>`

If run with `fix` then it changes those to:

```
      <span style="margin-right: -3px;
">(</span>
      <async-form-field-value
        :field="categoryField"
        class="byx-event-menu-category-field"
      ></async-form-field-value>
      <span style="margin-left: -3px;
">)</span>
```

Further examples
<img width="942" alt="screen shot 2018-06-27 at 23 42 44" src="https://user-images.githubusercontent.com/216041/42001352-084df7da-7a64-11e8-8af4-6bb207ee9c38.png">
<img width="947" alt="screen shot 2018-06-27 at 23 42 54" src="https://user-images.githubusercontent.com/216041/42001353-086a49c6-7a64-11e8-9864-e2aa68367278.png">

> Which rule, if any, is the bug related to?

`no-missing-end-of-source-newline`

> What CSS is needed to reproduce the bug?

```html
<template>
  <header>
    <b><sync-form-field-value
      :field="titleField"
      class="byx-event-menu-title-field"
    ></sync-form-field-value></b>
    <sync-form-field-value
      :field="dateTimeRangeField"
    ></sync-form-field-value>
    <sync-form-field-value
      v-if="ownerField.config.initialValue"
      :field="ownerField"
    ></sync-form-field-value>
    <div
      v-if="categoryField.config.initialValue"
      class="byx-actionmenu-category-label"
    >
      <span style="margin-right: -3px;">(</span>
      <async-form-field-value
        :field="categoryField"
        class="byx-event-menu-category-field"
      ></async-form-field-value>
      <span style="margin-left: -3px;">)</span>
    </div>
    <async-form-field-value
      v-if="patientField.config.initialValue"
      :field="patientField"
    ></async-form-field-value>
    <p v-if="smsField">
      <span>{{ i18n.sms_sent_1 }}</span> <span>{{ smsField }}</span>
    </p>
  </header>
</template>
```
> What stylelint configuration is needed to reproduce the bug?

```js
/**
 * @file Manages the root configuration settings for project wide stylelint.
 * @copyright Copyright (c) 2017-present, ProReNata AB
 * @module stylelint/root/configuration
 * @version 1.0.0
 * @since 0.0.2
 * @see {@link https://stylelint.io/} for further information.
 */

module.exports = {
  /**
   * Extend an existing configuration.
   * @type {String|Array.<String>}
   * @see {@link https://github.com/stylelint/stylelint/blob/master/docs/user-guide/configuration.md#extends}
   */
  extends: [
    /**
     * The standard shareable config for stylelint.
     * @type {string}
     * @see {@link https://github.com/stylelint/stylelint-config-standard}
     */
    'stylelint-config-standard',
    /**
     * Turns off all rules that are unnecessary or might conflict with prettier.
     * @type {string}
     * @see {@link https://github.com/shannonmoeller/stylelint-config-prettier}
     */
    'stylelint-config-prettier',
  ],

  /**
   * @type {array}
   */
  // plugins: [], // do not include empty

  /**
   * @type {array}
   */
  // processors: [], // do not include empty

  /**
   * @type {!Object}
   */
  rules: {
    'no-empty-source': null,
  },
};
```
> Which version of stylelint are you using?

```
    "postcss": "^6.0.23",
    "postcss-cssnext": "^3.1.0",
    "postcss-html": "^0.30.0",
    "postcss-import": "^11.1.0",
    "postcss-less": "^2.0.0",
    "postcss-loader": "^2.1.5",
    "postcss-safe-parser": "^3.0.1",
    "postcss-sass": "^0.3.2",
    "postcss-scss": "^1.0.6",
    "postcss-syntax": "^0.30.0",
    "postcss-url": "^7.3.2",
    "prettier": "^1.13.6",
    "stylelint": "^9.3.0",
    "stylelint-config-prettier": "^3.3.0",
    "stylelint-config-standard": "^18.2.0",
```
> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with `npm run` `"lint:css": "stylelint '**/*.{css,sass,scss,less,vue}'",`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Yes, `.vue` templates

> What did you expect to happen?

"No warnings to be flagged."

> What actually happened (e.g. what warnings or errors did you get)?

"The following warnings were flagged:"

```
Modules/Calendar/Components/EventMenuHeader.vue
 18:38  ✖  Unexpected missing end-of-source newline   no-missing-end-of-source-newline
 23:37  ✖  Unexpected missing end-of-source newline   no-missing-end-of-source-newline
```

**Comment by hudochenkov** [comment]
Closing as a duplicate. Answer: https://github.com/stylelint/stylelint/issues/3410#issuecomment-399658964.
